discord/bot.py

```diff
@@ -189,24 +189,40 @@ def get_application_command(
         Parameters
         ----------
         name: :class:`str`
-            The name of the command to get.
+            The qualified name of the command to get.
         guild_ids: List[:class:`int`]
             The guild ids associated to the command to get.
         type: Type[:class:`.ApplicationCommand`]
-            The type of the command to get. Defaults to :class:`.SlashCommand`.
+            The type of the command to get. Defaults to :class:`.ApplicationCommand`.
 
         Returns
         -------
         Optional[:class:`.ApplicationCommand`]
             The command that was requested. If not found, returns ``None``.
         """
-
-        for command in self._application_commands.values():
+        commands = self._application_commands.values()
+        for command in commands:
             if command.name == name and isinstance(command, type):
                 if guild_ids is not None and command.guild_ids != guild_ids:
                     return
                 return command
 
+        # If a top-level command was not found, attempt to search by qualified command name
+        if name.lower() == name and 1 < len(names := name.split()) <= 3 and all(names):
```

Comment:
I like the idea but not the logic here, can't we remove this part and just walk through subcommands above?

Reply:
Perhaps something like this then?
```py
        for command in commands:
            if command.name == name and isinstance(command, type):
                ...
            elif (names := name.split())[0] == name and isinstance(command, SlashCommandGroup):
                while len(names) > 1:
                   ...
```